### ISSUE TITLE
Doubleslash in url

### ISSUE THREAD

**Issue by malx122** [Starcounter/Starcounter#3662]
In most web places double slash works in url without problem. (same as a single slash)
Some examples:
```
http://www.dn.se/sport/
http://www.dn.se//sport/
http://www.dn.se/\sport/
```
The urls all above works.
But doing the same against a SC server will make the result "not found". This is quite sensitive since this situation can happen a lot when combining user settings urls with server settings. Some put a trailing slash, some will not and when combining them you will either get double slash or no slash. The prefered way is to make sure there is at least one slash which can give the double slash situtation.

So it would be good if that can be fixed. (I understand it can be somewhat hard with those url parameter and stuff?)

**Comment by per-samuelsson** [comment]
Thanks for pointing this out to us, @malx122.

I guess fixing it would require us to extend the URI matcher, right @alemoi, @chrhol?

**Comment by miyconst** [comment]
@malx122 this is not a good idea to use double-slashes in the URL path. More reading is available on Stackoverflow:

>It looks like current versions of Apache and IIS both will ignore the extra slashes while resolving the path and return the document that would have been returned had the URL not had extra slashes. However, browsers (I tested IE 8 and Chrome 9) get confused by any relative URLs (containing parent path components) of resources in the page, which produces bad results.

- [Are there any downsides to using double-slashes in URLs?](http://stackoverflow.com/questions/11256190/are-there-any-downsides-to-using-double-slashes-in-urls).
- [Is a URL with // in the path-section valid?](http://stackoverflow.com/questions/20523318/is-a-url-with-in-the-path-section-valid).
- [What does the double slash mean in URLs?](http://webmasters.stackexchange.com/questions/8354/what-does-the-double-slash-mean-in-urls/8381#8381).

In short: Apache & IIS has build in feature to treat `//` as `/`, but browsers do not correctly resolve virtual paths relative to double-slash-ed.

It is very easy to make a helper method which would make sure that all of your URLs are combined correctly. Take a look at [System.Uri.TryCreate](https://msdn.microsoft.com/en-us/library/system.uri.trycreate.aspx).

If you really want to, you can add a request filter, which would transfer your URLs, but be aware that you can never be sure that your filter will be executed before filters from other applications.

```cs
Application.Current.Use((request, response) => {
    if (request.Uri.IndexOf("//") >= 0) {
        string uri = request.Uri.Replace("//", "/");
        return Self.GET(uri);
    }

    return null;
});
```

**Comment by malx122** [comment]
Ok, means that I will need to find out a way to do it with check for this. (Like I write above, this is not a result of deliberate double slashes, it's because it is user given urls that being combined with relative adresses). But it might not be so hard, more that I need to check it everywhere :).
